rct229/rules/section21/section21rule6.py

```diff
@@ -0,0 +1,120 @@
+        #  is ready.
+        baseline_system_types = {
+            "SYS-7A": ["hvac_sys_7_a"],
+            "SYS-11A": ["hvac_sys_11_a"],
+        }
+        # if any system type found in the APPLICABLE_SYS_TYPES then return applicable.
+        return any(
+            [key in APPLICABLE_SYS_TYPES for key in baseline_system_types.keys()]
+        )
+
+    def list_filter(self, context_item, data):
+        fluid_loop_b = context_item.baseline
+        loop_boiler_dict = data["loop_boiler_dict"]
+        # Heating loop and have two boilers.
+        return (
+            getattr_(fluid_loop_b, "FluidLoop", "type") == FLUID_LOOP.HEATING
+            and len(loop_boiler_dict[fluid_loop_b["id"]]) == 2
+        )
+
+    class HeatingFluidLoopRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section21Rule6.HeatingFluidLoopRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+            )
+
+        def get_calc_vals(self, context, data=None):
+            fluid_loop_b = context.baseline
+            boiler_list = data["loop_boiler_dict"][fluid_loop_b["id"]]
+            # Guarantee two boilers in this list.
+            boiler_1 = boiler_list[0]
+            boiler_2 = boiler_list[1]
+
+            return {
+                "boiler_1_operation_lower_limit": getattr_(
+                    boiler_1, "boiler", "operation_lower_limit"
+                ),
+                "boiler_1_operation_upper_limit": getattr_(
+                    boiler_1, "boiler", "operation_upper_limit"
+                ),
+                "boiler_1_rated_capacity": getattr_(
+                    boiler_1, "boiler", "rated_capacity"
+                ),
+                "boiler_2_operation_lower_limit": getattr_(
+                    boiler_2, "boiler", "operation_lower_limit"
+                ),
+                "boiler_2_operation_upper_limit": getattr_(
+                    boiler_2, "boiler", "operation_upper_limit"
+                ),
+                "boiler_2_rated_capacity": getattr_(
+                    boiler_2, "boiler", "rated_capacity"
+                ),
+            }
+
+        def rule_check(self, context, calc_vals=None, data=None):
+            boiler_1_operation_lower_limit = calc_vals["boiler_1_operation_lower_limit"]
+            boiler_1_operation_upper_limit = calc_vals["boiler_1_operation_upper_limit"]
+            boiler_1_rated_capacity = calc_vals["boiler_1_rated_capacity"]
+            boiler_2_operation_lower_limit = calc_vals["boiler_2_operation_lower_limit"]
+            boiler_2_operation_upper_limit = calc_vals["boiler_2_operation_upper_limit"]
+            boiler_2_rated_capacity = calc_vals["boiler_2_rated_capacity"]
+            return (
+                boiler_1_operation_lower_limit == 0
+                and boiler_1_operation_upper_limit == boiler_1_rated_capacity
+                and boiler_2_operation_lower_limit == boiler_2_rated_capacity
```

Comment:
I question whether the logic is correct in the RDS.  Unless I am not understanding, which is likely, shouldn't the logic be:
```
and boiler_2_operation_lower_limit == boiler_1_rated_capacity
and boiler_2_operation_upper_limit == boiler_1_rated_capacity + boiler_2_rated_capacity
```
?

Reply:
I do have the same question, I am reserving this PR for this Thursday RDS meeting.

Reply:
change to 
`
            - Case 1: If boiler_1's operation lower limit is 0 and operation upper limit is equal to its rated capacity, and boiler_2's operation lower limit is equal to boiler_1 rated capacity and operation upper limit is equal to the sum of boiler 1 and boiler 2 rated capacity: `if ( boiler_1.operation_lower_limit == 0 ) AND ( boiler_1.operation_upper_limit == boiler_1.rated_capacity ) AND ( boiler_2.operation_lower_limit == boiler_1.rated_capacity ) AND ( boiler_2.operation_upper_limit == boiler_21.rated_capacity + boiler_2.rated_capacity ): PASS`

            - Case 2: Else if boiler_2's operation lower limit is 0 and operation upper limit is equal to its rated capacity, and boiler_1's operation lower limit is equal to boiler_2 rated capacity and operation upper limit is equal to the sum of boiler 1 and boiler 2 rated capacity: `if ( boiler_2.operation_lower_limit == 0 ) AND ( boiler_2.operation_upper_limit == boiler_2.rated_capacity ) AND ( boiler_1.operation_lower_limit == boiler_2.rated_capacity ) AND ( boiler_1.operation_upper_limit == boiler_1.rated_capacity + boiler_2.rated_capacity ): PASS`
`